src/D2L.Bmx/UnixParentProcess.cs

```diff
@@ -0,0 +1,47 @@
+using System.Diagnostics;
+
+namespace D2L.Bmx;
+
+internal class UnixParentProcess {
+
+	public static string GetParentProcessName() {
+		int parentPid = GetParentProcessId( GetParentProcessId( Process.GetCurrentProcess().Id ) );
```

Comment:
do you see an analyzer suggestion on this?
```suggestion
		int parentPid = GetParentProcessId( GetParentProcessId( Environment.ProcessId ) );
```
also - why two `GetParentProcessId` calls?

Reply:
vscode did not suggest that but I can change to what you suggested

Reply:
I think there could be a big tree of potential processes before getting to the shell if something else calls bmx so I'm not sure how rigid this will be unless we want to keep going up trying to find something. For example maybe aws credentials-process would break this and instead of bash being the 2nd parent its some aws process

Reply:
> The first parent is bmx and then we have to go up one more parent to get the shell

This doesn't sound right. I tested this branch in a PowerShell session launched by Bash, and it defaulted to Bash not PowerShell. I don't think the two get parent calls are necessary.

> I think there could be a big tree of potential processes before getting to the shell if something else calls bmx so I'm not sure how rigid this will be unless we want to keep going up trying to find something. For example maybe aws credentials-process would break this and instead of bash being the 2nd parent its some aws process

We only care if the user is launching bmx from within their shell directly. If there's an intermediate process, then the shell is irrelevant.

Reply:
Similarly, when I launched bash from PowerShell then launch bmx, it defaulted to PowerShell.

Reply:
To confirm - you're running the published native AOT build, not `dotnet run` or launching via IDE?

Reply:
Yep I'm using published build. My process is 
`dotnet publish osx-x64 -o out`
`$(./out/bmx print)`